pkg/app/target.go

```diff
@@ -96,6 +95,26 @@ func (a *App) DeleteTarget(ctx context.Context, name string) error {
 	return nil
 }
 
+// UpdateTargetConfig updates the subscriptions for an existing target
+func (a *App) UpdateTargetSubscription(ctx context.Context, name string, subs []string) error {
+	a.configLock.Lock()
+	for _, subName := range subs {
+		if _, ok := a.Config.Subscriptions[subName]; !ok {
+			return fmt.Errorf("subscription %q does not exist", subName)
```

Comment:
The config lock will not be unlocked if this error is returned.

Reply:
woops.. started with a defer in a previous iteration and ended up changing the logic.  nice catch!